rule/expr.go

```diff
@@ -67,29 +77,30 @@ func (n *exprNot) Eval(params Params) (*Value, error) {
 	return BoolValue(true), nil
 }
 
-// Contract returns the Contract for exprNot, and makes it comply with the TypedExpression interface.
-func (n *exprNot) Contract() Contract {
-	return Contract{
-		ReturnType: BOOLEAN,
-		Terms:      []Term{{Type: BOOLEAN, Cardinality: ONE}},
-	}
-}
-
 type exprOr struct {
 	operator
 }
 
-// Or creates an expression that takes at least two operands and evaluates to true if one of the operands evaluates to true.
-// All the given operands must evaluate to a boolean.
-func Or(v1, v2 Expr, vN ...Expr) Expr {
```

Comment:
We used to have a compile-time arity check but now we only have a runtime one, what is your reasoning about this?

Reply:
Yup, I did explicitly think about this.  It's a shame to lose the compile time check, but it complicates the implementation of these public functions and is, in fact an incomplete arity check for any type that has the `vN ...Expr` form.   For forms that don't have a variable number of operands, it's easy to maintain the compile time arity check, but for cases like this we'd have to push `v1`, `v2` and every element of `vN` onto a slice and then reexpand it inline when calling `consumeOperands`.   In an ideal world Go's compiler would let you do this:

`o.consumerOperands(v1, v2, ...vN)`